src/main/java/frc/team166/robot/subsystems/Lift.java

```diff
@@ -0,0 +1,184 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2017-2018 FIRST. All Rights Reserved.                        */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+/*Functions 
+read encoder 
+set motor to raise lift 
+set motor to lower lift 
+shift into low or high gear 
+read top limit switch 
+read bottom limit switch 
+read other limit switches...? 
+ go to specified height
+
+Commands 
+go to specified height 
+make the robot climb 
+raise lift 
+lower lift 
+*/
+
+package frc.team166.robot.subsystems;
+
+import javax.lang.model.util.ElementScanner6;
+
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonSRX;
+
+import edu.wpi.first.wpilibj.Encoder;
+import edu.wpi.first.wpilibj.DoubleSolenoid;
+import edu.wpi.first.wpilibj.SpeedControllerGroup;
+import edu.wpi.first.wpilibj.DoubleSolenoid.Value;
+import edu.wpi.first.wpilibj.command.Command;
+import edu.wpi.first.wpilibj.command.Subsystem;
+import frc.team166.chopshoplib.commands.ActionCommand;
+import frc.team166.chopshoplib.commands.CommandChain;
+import frc.team166.chopshoplib.commands.SubsystemCommand;
+import frc.team166.robot.RobotMap;
+import frc.team166.robot.RobotMap.Encoders;
+import edu.wpi.first.wpilibj.DigitalInput;
+
+/** 
+ * An example subsystem.  You can replace me with your own Subsystem. 
+ */
+public class Lift extends Subsystem {
+    // Defines Limit Switches (Digital imputs)
+    DigitalInput TopLimitSwitchLift = new DigitalInput(RobotMap.DigitalImputs.LIMIT_SWITCH_LIFT_BOTTOM);
+    DigitalInput BottomLimitSwitchLift = new DigitalInput(RobotMap.DigitalImputs.LIMIT_SWITCH_LIFT_TOP);
+    // Defines Encoders 
+    Encoder liftEncoder = new Encoder(RobotMap.Encoders.LIFT_A, RobotMap.Encoders.LIFT_B);
```

Comment:
You should be using a consistent naming scheme within a class: why do some things have "lift" in the name, while others don't? Personally I would leave out "lift" from anything already inside the Lift subsystem. Access these things from anywhere outside this file would look like this: "Robot.Lift.TopLimitSwitch" or "Robot.Lift.Encoder". There is really no need to keep saying "lift" in all these variables.